browser/ui/webui/brave_wallet/wallet_panel_ui_browsertest.cc

```diff
@@ -5,66 +5,296 @@
 #include "chrome/browser/ui/browser.h"
 #include "chrome/browser/ui/browser_tabstrip.h"
 #include "chrome/common/chrome_isolated_world_ids.h"
 #include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/javascript_dialogs/app_modal_dialog_controller.h"
+#include "components/javascript_dialogs/app_modal_dialog_view.h"
+#include "components/javascript_dialogs/tab_modal_dialog_manager.h"
+#include "components/web_modal/web_contents_modal_dialog_manager.h"
 #include "content/public/test/browser_test.h"
 #include "content/public/test/test_utils.h"
+#include "services/network/public/cpp/weak_wrapper_shared_url_loader_factory.h"
+#include "services/network/test/test_url_loader_factory.h"
 
-class WalletPanelUIBrowserTest : public InProcessBrowserTest {
- public:
-  void SetUp() override {
-    feature_list_.InitAndEnableFeature(
-        brave_wallet::features::kNativeBraveWalletFeature);
-    InProcessBrowserTest::SetUp();
+using content::EvalJsResult;
+
+namespace {
+
+std::string SelectInNetworkList(const std::string& selector) {
+  return base::StringPrintf("window.testing.walletNetworks.querySelector(`%s`)",
+                            selector.c_str());
+}
+
+std::string SelectInAddNetworkDialog(const std::string& selector) {
+  return base::StringPrintf(
+      "window.testing.addWalletNetworkDialog.querySelector(`%s`)",
+      selector.c_str());
+}
+
+std::string DoubleClick() {
+  return ".dispatchEvent((function (){const e = "
+         "document.createEvent('MouseEvents');e.initEvent('dblclick',true,true)"
+         ";return e;})())";
+}
+
+std::string CeloNetwork() {
+  return "[data-test-chain-id='chain-0xa4ec']";
+}
+
+std::string PolygonNetwork() {
+  return "[data-test-chain-id='chain-0x89']";
+}
+
+std::string NetworkNameSpan() {
+  return "[class|='NetworkName']";
+}
+
+std::string CeloNetworkHideButton() {
+  return CeloNetwork() + " .hide-network-button";
+}
+
+std::string CeloNetworkChainName() {
+  return CeloNetwork() + " .chainName";
+}
+
+std::string NetworksButton() {
+  return R"([data-test-id='select-network-button'])";
+}
+
+std::string Select(const std::string& selector) {
+  return base::StringPrintf(R"(document.querySelector(`%s`))",
+                            selector.c_str());
+}
+
+std::string Select(const std::string& selector1, const std::string& selector2) {
+  return base::StringPrintf(R"(document.querySelector(`%s %s`))",
+                            selector1.c_str(), selector2.c_str());
+}
+
+void NonBlockingDelay(const base::TimeDelta& delay) {
+  base::RunLoop run_loop(base::RunLoop::Type::kNestableTasksAllowed);
+  base::ThreadTaskRunnerHandle::Get()->PostDelayedTask(
+      FROM_HERE, run_loop.QuitWhenIdleClosure(), delay);
+  run_loop.Run();
+}
+
+bool WaitFor(content::WebContents* web_contents, const std::string& selector) {
+  for (int i = 0; i < 1000; ++i) {
+    if (EvalJs(web_contents, "!!(" + selector + ")").ExtractBool())
```

Comment:
Would using `content::ExecuteScriptAndExtractBool` remove the need to remove CSP?

Reply:
Technically yes, but `ExecuteScriptAndExtract` family is cumbersome and deprecated in favor of `EvalJs`. With workaround wallet and settings ui are testable without CSP.
https://source.chromium.org/chromium/chromium/src/+/main:content/public/test/browser_test_utils.h;l=538?
q=ExecuteScriptAndExtractBool&ss=chromium